tests/fixtures/webap.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+import re
+import requests
+# Disable warnings from urllib, since there will be lots about insecure connections
+#  given that we're using a self-signed cert for the server in the test environment
+requests.packages.urllib3.disable_warnings()
+
+import selenium
+import selenium.webdriver
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.remote.webelement import WebElement
+
+from util.config import Config
+from util.rkauth_client import rkAuthClient
+
+@pytest.fixture( scope='session' )
+def webap_url():
+    return Config.get().value( 'webap.webap_url' )
+
+@pytest.fixture
+def webap_rkauth_client( webap_url, user ):
+    client = rkAuthClient( webap_url, 'test', 'test_password', verify=False )
+    client.verify_logged_in()
+    return client
+
+@pytest.fixture
+def webap_admin_client( webap_url, admin_user ):
+    client = rkAuthClient( webap_url, 'admin', 'admin', verify=False )
+    client.verify_logged_in()
+    return client
+
+@pytest.fixture
+def webap_browser_logged_in( browser, user ):
+    cfg = Config.get()
+    webap_url = cfg.value( 'webap.webap_url' )
+    username = cfg.value( 'conductor.username' )
+    password = cfg.value( 'conductor.password' )
+    browser.get( webap_url )
+    input_password = WebDriverWait( browser, timeout=5 ).until( lambda d: d.find_element( By.ID, 'login_password' ) )
+    input_password.clear()
+    input_password.send_keys( password )
+    input_user = browser.find_element( By.ID, 'login_username' )
+    input_user.clear()
+    input_user.send_keys( username )
+    buttons = browser.find_elements( By.TAG_NAME, 'button' )
+    button = None
+    for possible_button in buttons:
+        if possible_button.get_attribute( "innerHTML" ) == "Log In":
+            button = possible_button
+            break
+    assert button is not None
+    button.click()
+
+    def check_logged_in( d ):
```

Comment:
Note sure if the variable `d` is used here, but I may be missing the reason is it still worth having passed to the function

Reply:
It's a callback thing in `WebDriverWait.until`.  Even though I don't use it, the function passed to `until` will be getting an argument when `until` calls it.